edx/edx-enterprise
# ENT-649 Only attempt to transmit SAP SuccessFactors details for which a currently-linked SSO user exists

**Description:**

This pull request changes the user data transmission routine to only attempt to transmit and log user details in SAP SuccessFactors for users who are currently linked via SSO to the relevant provider; if there isn't a linked provider, then there's no real way for us to transmit data back anyway.

**JIRA:** [ENT-649](https://openedx.atlassian.net/browse/ENT-649)

**Dependencies:** None

**Merge deadline:** 9 October 2017

**Installation instructions:** N/A

**Testing instructions:**

1. Create ECU linked to an EnterpriseCustomer set up for SAPSF (real details not necessary)
2. Run data transmission management command
    - Note that no data transmision is attempted
3. Log into that user, and authenticate with the SSO provider linked to the EC
4. Run data transmission management command
    - Note that data transmission is attempted on that user (verify by checking the data transmission audit table)

To test improved login requirement robustness:

1. While fully logged-out, navigate to a program or course enterprise enrollment page.
2. Rather than logging in with SSO, as prompted, navigate directly to the platform login page and log in.
3. Again, navigate directly to a program or course enterprise enrollment page
4. Verify that your dropped back to an SSO-based login, rather than the current state of being allowed to enroll in the course/program.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.
